tests/sklearn/test_sklearn_models.py

```diff
@@ -462,10 +496,18 @@ def check_offset(model_class, n_bits, x, y):
         model.fit(x, y)
 
 
-def check_subfunctions(fitted_model, model_class, x):
-    """Check subfunctions."""
+def check_inference_methods(model, model_class, x, check_float_array_equal):
```

Comment:
this test now not only tests that all inference methods (predict, predict_proba, decision_function) work but also makes sure results (including shapes) are coherent between `fhe="disable"` and `fhe="simulate"`, which was not done before